app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -26,12 +28,60 @@ def get_middleware_servers
           @ems.eaps(feed).map do |eap|
             @eaps << eap
             server = parse_middleware_server(eap)
+
+            machine_id = @ems.machine_id(eap.feed)
+            host_instance = find_host_by_bios_uuid(machine_id)
+            if host_instance.nil?
+              host_instance = find_host_by_bios_uuid(alternate_machine_id(machine_id))
+            end
+
+            unless host_instance.nil?
+              server[:lives_on_id] = host_instance.try(:id)
+              server[:lives_on_type] = host_instance.try(:type)
+            end
+
             @data[:middleware_servers] << server
             @data_index.store_path(:middleware_servers, :by_nativeid, server[:nativeid], server)
           end
         end.flatten
       end
 
+      def alternate_machine_id(machine_id)
+        # see BZ #1294461 : here, we'll try to adjust the machine ID to the format from that bug
```

Comment:
Doesnt your excellent explanation cover everything?
If you refer to a BZ, which in this case even has redhat internal information, I'd use a proper link.